### PR TITLE
Fix/update python

### PR DESCRIPTION
**Pull request by yuanhengzhao**
In this pr, I upgrade the python version of billing-api from 2 to 3; fix all the errors, warnings and test failures due to the upgrade;

Update a lot of dependencies for both billing-api and billing-ui;

Put all config in one globalConfig.json file and ignore it so that future developer can put the sensitive configs in there without pushing those configs accidently

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,23 +31,77 @@ Pull the latest image:
...
+                      "period_end": "2017-05",
+                      "discount": 0.9
+                  },
+                  {
+                      "period_start": "2017-06",
+                      "period_end": "2017-06",
+                      "discount": 0.7
+                  }
+              ],
+              "oicr_demo_dusan": [
```
Comment (andricDu) [inline README.md]: remove the name here

File: README.md [hunk 2]
```diff
@@ -31,23 +31,77 @@ Pull the latest image:
...
+              {
+                  "period_start": "2016-11-03",
+                  "period_end": "2016-12-22",
+                  "cpu_price": 0.06,
+                  "volume_price": 0.03,
+                  "image_price": 0.03
+              }
+          ],
+          "DISCOUNTS": {
+              "oicr_demo_rahul": [
```
Comment (andricDu) [inline README.md]: remove this name as well

File: README.md [hunk 3]
```diff
@@ -31,23 +31,77 @@ Pull the latest image:
...
+              "monthly",
+              "yearly"
+          ], // Bucketing options for query.
+          "FLASK_LOG_FILE": "./logs/billing.log",
+          "BILLING_ROLE": "billing_test",
+          "INVOICE_ROLE": "invoice_test",
+          "OICR_ADMIN": "oicr_admin",
+          "OICR_ADMINS": [
+              "Rahul.Verma@oicr.on.ca",
+              "Sid.Joshi@oicr.on.ca"
```
Comment (andricDu) [inline README.md]: remove

File: README.md [hunk 4]
```diff
@@ -31,23 +31,77 @@ Pull the latest image:
...
+              "weekly",
+              "monthly",
+              "yearly"
+          ], // Bucketing options for query.
+          "FLASK_LOG_FILE": "./logs/billing.log",
+          "BILLING_ROLE": "billing_test",
+          "INVOICE_ROLE": "invoice_test",
+          "OICR_ADMIN": "oicr_admin",
+          "OICR_ADMINS": [
+              "Rahul.Verma@oicr.on.ca",
```
Comment (andricDu) [inline README.md]: remove

File: billing-api/billing_server/billing/usage_queries.py [hunk 5]
```diff
@@ -26,12 +26,12 @@ def __init__(self, database_url, logger, billing_role='billing', initialized=Tru
         self.database = records.Database(database_url)
         logger.info('Successfully connected to database')
         self.user_map = {}
-        if initialized:
-            self.refresh_user_id_map()
+        if (initialized):
```
Comment (andricDu) [inline billing-api/billing_server/billing/usage_queries.py]: remove these parenthesis

File: billing-api/billing_server/billing/usage_queries.py [hunk 6]
```diff
@@ -26,12 +26,12 @@ def __init__(self, database_url, logger, billing_role='billing', initialized=Tru
...
+          self.refresh_user_id_map()
 
     def close(self):
         if hasattr(self, 'database'):
-            self.database.close()
+          self.database.close()
```
Comment (andricDu) [inline billing-api/billing_server/billing/usage_queries.py]: python should be using indentation of 4 spaces, please revert this

File: billing-api/billing_server/billing/usage_queries.py [hunk 7]
```diff
@@ -26,12 +26,12 @@ def __init__(self, database_url, logger, billing_role='billing', initialized=Tru
         self.database = records.Database(database_url)
         logger.info('Successfully connected to database')
         self.user_map = {}
-        if initialized:
-            self.refresh_user_id_map()
+        if (initialized):
+          self.refresh_user_id_map()
```
Comment (andricDu) [inline billing-api/billing_server/billing/usage_queries.py]: python should be using indentation of 4 spaces, please revert this

File: billing-api/requirements.txt [hunk 8]
```diff
@@ -1,43 +1,46 @@
...
-oslo.utils==3.16.0
-pbr==1.10.0
-positional==1.1.1
-pycparser==2.14
-pyparsing==2.1.9
-pytest==3.0.4
-python-dateutil==2.5.3
-python-keystoneclient==3.5.0
-pytz==2016.7
-records==0.4.3
```
Comment (andricDu) [inline billing-api/requirements.txt]: you removed this dependency yet you still require it in the usage_queries.py
This is what you get if you don't have this:

```bash
Traceback (most recent call last):
  File "/home/xxx/Documents/workspace/billing/billing-api/run.py", line 16, in <module>
    from billing_server.billing import app
  File "/home/xxx/Documents/workspace/billing/billing-api/billing_server/billing/__init__.py", line 28, in <module>
    from .usage_queries import Collaboratory
  File "/home/xxx/Documents/workspace/billing/billing-api/billing_server/billing/usage_queries.py", line 16, in <module>
    import records
ModuleNotFoundError: No module named 'records'
```

File: billing-api/billing_server/billing/config/default.py [hunk 9]
```diff
@@ -13,36 +13,26 @@
...
-#FLASK_LOG_FILE = '/srv/billing-api/logs/billing.log'
-FLASK_LOG_FILE = './logs/billing.log'
-BILLING_ROLE = 'billing_test'
-#INVOICE_ROLE = 'invoice'
-INVOICE_ROLE = 'invoice_test'
-OICR_ADMIN = 'oicr_admin'
+import json
+
+config={}
+with open('../globalConfig.json') as json_file: # This file is under billing/ folder
```
Comment (andricDu) [inline billing-api/billing_server/billing/config/default.py]: did you confer with @jmimico that he overwrites this default.py file when running the docker image? Otherwise this won't work.

File: .gitignore [hunk 10]
```diff
@@ -200,4 +200,7 @@ invoice-api/node_modules/
 .DS_Store
 
 #rsa
-rsa
\ No newline at end of file
+rsa
+
+#configuration
+globalConfig.json
```
Comment (andricDu) [inline .gitignore]: if you are going to want a file for globalConfig, you should at least provide an example version with `globalConfig.json.example`

File: billing-ui/package.json [hunk 11]
```diff
@@ -40,13 +40,13 @@
     "gzip-size": "3.0.0",
     "html-webpack-plugin": "2.22.0",
     "json-loader": "0.5.4",
-    "node-sass": "^3.13.1",
+    "node-sass": "^4.12.0",
```
Comment (andricDu) [inline billing-ui/package.json]: This is a PR for fixing the python version, why are we updating UI dependencies?

File: billing-ui/package.json [hunk 12]
```diff
@@ -59,6 +59,7 @@
     "encode-uri-query": "^1.0.1",
     "gsap-promise": "2.0.0",
     "highcharts": "^5.0.14",
+    "jquery": "^3.4.1",
```
Comment (andricDu) [inline billing-ui/package.json]: why was this added?

**Comment by yuanhengzhao** [comment]
Remove all unrelated updates for biling-ui.
Modify Dockerfile for python3.

**Comment by andricDu** [comment]
Thanks!
